circleguard/comparer.py

```diff
@@ -144,7 +146,32 @@ def _compare_two_replays(replay1, replay2):
         if (Mod.HR in replay1.mods) ^ (Mod.HR in replay2.mods):
             xy1[:, 1] = 384 - xy1[:, 1]
 
-        return Comparer._compute_data_similarity(xy1, xy2)
+        play_sequence_1 = xy1.T.copy() # Need copy, as the cross_correlate method uses an in-place modification of the matrix
+        play_sequence_2 = xy2.T.copy()
+
+        # Sectioned into 20 chunks, used to ignore outliers (eg. long replay with breaks is copied, and the cheater cursordances during the break)
+        horizontal_length = play_sequence_1.shape[1] - play_sequence_1.shape[1] % num_chunks
+        play_sequence_1_sections = np.hsplit(play_sequence_1[:,:horizontal_length], num_chunks)
+        play_sequence_2_sections = np.hsplit(play_sequence_2[:,:horizontal_length], num_chunks)
+        correlations = []
+        for (play_sequence_1, play_sequence_2) in zip(play_sequence_1_sections, play_sequence_2_sections):
+            cross_correlation_matrix = Comparer.find_cross_correlation(play_sequence_1, play_sequence_2)
+            #Pick the lag with the maximum correlation, this likely in most cases is 0 lag
+            max_correlation = max(cross_correlation_matrix.reshape(cross_correlation_matrix.size))
+            correlations.append(max_correlation)
+        average_distance = Comparer._compute_data_similarity(xy1, xy2)
+        # Out of all the sections, we should have 20 sections, we pick the one with the median correlation, so we throw away any outliers
+        median_correlation = np.median(np.array(sorted(correlations)))
+        return (average_distance, median_correlation)
+
+    @staticmethod
+    def find_cross_correlation(play_sequence_1, play_sequence_2):
```

Comment:
rename to `xy1` and `xy2`

`_compute_data_similarity` params should also be renamed to this